src/osdc/Objecter.cc

```diff
@@ -1208,18 +1208,29 @@ void Objecter::handle_osd_map(MOSDMap *m)
 
 	cluster_full = cluster_full || _osdmap_full_flag();
 	update_pool_full_map(pool_full_map);
+
+	// check all outstanding requests on every epoch, including need_resend
+	for (auto p = need_resend.begin(); p != need_resend.end(); ) {
```

Comment:
it's a loop.. the bug comes up when we get multiple maps at once and fail to _calc_target on subsequent maps